src/impl/icetransport.cpp

```
@@ -140,6 +140,10 @@ IceTransport::IceTransport(const Configuration &config, candidate_callback candi
 			addIceServer(server);
 }
 
+void IceTransport::setIceAttributes(string uFrag, string pwd) {
+	juice_set_local_ice_attributes(mAgent.get(), uFrag.c_str(), pwd.c_str());
```

You should throw an `std::invalid_argument("Invalid ICE attributes")` exception on failure otherwise it may silently fail.

---

Done!